### PR TITLE
Add delegation for settings

### PR DESCRIPTION
**Pull request by TymanWasTaken**
Usage:
```kt
var settingName: SettingType by settings.delegate(defaultValue)
```

### CODE REVIEW

File: Aliucord/src/main/java/com/aliucord/SettingsUtilsJSON.kt [hunk 1]
```diff
@@ -8,6 +8,8 @@ import java.io.File
 import java.lang.reflect.Type
 import java.math.BigDecimal
 import java.util.*
+import kotlin.reflect.KClass
+import kotlin.reflect.typeOf
```
Comment (DiamondMiner88) [inline Aliucord/src/main/java/com/aliucord/SettingsUtilsJSON.kt:12]: Unused import probably

**Comment by TymanWasTaken** [comment]
Review when

**Comment by mantikafasi** [comment]
never

File: Aliucord/src/main/java/com/aliucord/api/SettingsAPI.java [hunk 2]
```diff
@@ -203,4 +203,33 @@ public void setString(String key, String val) {
...
+     */
+    public Object getUnknown(String key, Object defValue) {
+        Object value = null;
+        if (defValue instanceof String) value = getString(key, (String) defValue);
+        else if (defValue instanceof Boolean) value = getBool(key, (Boolean) defValue);
+        else if (defValue instanceof Long) value = getLong(key, (Long) defValue);
+        else if (defValue instanceof Float) value = getFloat(key, (Float) defValue);
+        else if (defValue instanceof Integer) value = getInt(key, (Integer) defValue);
+        else value = getObject(key, defValue);
+        return value;
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/api/SettingsAPI.java:219]: get rid of the value variable and instead return right away, then you can also drop all the else

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Review by mantikafasi — APPROVED** [review]

**Review by Vendicated — APPROVED** [review]

**Pull request merged**